src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1294,9 +1294,17 @@ internal static object[] GetCustomAttributes(RuntimeType type, RuntimeType caTyp
                 type = type.BaseType as RuntimeType;
             }
 
-            object[] typedResult = CreateAttributeArrayHelper(arrayType, result.Count);
-            Array.Copy(result.ToArray(), 0, typedResult, 0, result.Count);
-            return typedResult;
+            if (result.Count > 0)
+            {
+                object[] typedResult = CreateAttributeArrayHelper(arrayType, result.Count);
+                for (var i = 0; i < result.Count; i++)
+                {
+                    typedResult[i] = result[i];
+                }
+                return typedResult;
+            }
+            // Return a cached, empty array
```

Comment:
Note: I had to look at this twice - may be confusing to look at in text here: there are 2 instances of this helper. The one in `CustomAttribute` is already `object[]`, the one on `Attribute` is `Attribute[]` (and needs to be, since the return there is more specific all the way up).